src/components/include/protocol_handler/protocol_handler.h

```diff
@@ -134,11 +134,12 @@ class ProtocolHandler {
    * @param rejected_params list of parameters name that are rejected.
    * Only valid when generated_session_id is 0. Note, even if
    * generated_session_id is 0, the list may be empty.
+   * @param err_reason string with NACK reason. Only valid when
+   * generated_session_id is 0.
    */
-  virtual void NotifySessionStarted(
-      const SessionContext& context,
-      std::vector<std::string>& rejected_params,
-      const std::string err_reason = std::string()) = 0;
+  virtual void NotifySessionStarted(const SessionContext& context,
+                                    std::vector<std::string>& rejected_params,
```

Comment:
@ychernysheva const

Reply:
It seems that I cannot do that because of https://github.com/LuxoftSDL/sdl_core/blob/develop/src/components/protocol_handler/src/protocol_handler_impl.cc#L1928
We need to have an opportunity to modify this vector through adding new reasons if exist